lua/nvim-treesitter/health.lua

```diff
@@ -0,0 +1,47 @@
+local api = vim.api
+
+local install = require'nvim-treesitter.install'
+local queries = require'nvim-treesitter.query'
+local locals = require'nvim-treesitter.locals'
+
+local health_start = vim.fn["health#report_start"]
+local health_ok = vim.fn['health#report_ok']
+local health_info = vim.fn['health#report_info']
+local health_warn = vim.fn['health#report_warn']
+local health_error = vim.fn['health#report_error']
+
+local M = {}
+
+-- TODO(vigoux): Maybe we should move each check to be perform in its own module
+function M.checkhealth()
+  -- Installation dependency checks
+  health_start('Installation')
+  install.checkhealth()
+
+  local missing_parsers = {}
+  -- Parser installation checks
+  for parser_name, repo in pairs(install.repositories) do
+    local installed = #api.nvim_get_runtime_file('parser/'..parser_name..'.so', false)
+
+    -- Only print informations about installed parsers
+    if installed == 1 then
+      health_start(parser_name .. " parser healthcheck")
+      health_ok(parser_name .. " parser found.")
+
+      locals.checkhealth(parser_name)
+    elseif installed > 1 then
+      health_warn(string.format("Multiple parsers found for %s, only %s will be used.", parser_name, installed[1]))
```

Comment:
could this happen ? i don't yet have documentation for nvim_get_runtime_file, i don't quite remember how it behaves, could you enlighten me please ?

Reply:
I think that `nvim_get_runtime_file` will actually return you the list of the files that match the regex.
I don't know if it could happne though but my idea was that if someone installs a C parser in addition to builtin one, a wargning could be send